server/api/crud/artifacts.py

```diff
@@ -47,6 +50,23 @@ def store_artifact(
                 f"Artifact with conflicting project name - {data['project']} while request project : {project}."
                 f"key={key}, uid={uid}, data={data}"
             )
+
+        # calculate the size of the artifact if needed
+        if calc_size:
+            if "spec" in data and "size" not in data["spec"]:
+                if "target_path" in data["spec"]:
+                    path = data["spec"].get("target_path")
+                    try:
+                        file_stat = (
+                            server.api.crud.Files().get_filestat_with_project_secrets(
+                                auth_info, path=path
+                            )
+                        )
+                        data["spec"]["size"] = file_stat["size"]
+                    except mlrun.errors.MLRunNotFoundError:  # if the path was not found the size will be N/A
+                        logger.debug("Path was not found", path=path)
+                        pass
```

Comment:
I think this should always happen. anyway it is wrapped with try/except so worst case scenario, the artifact size would be N/A.
also, worth putting entire logic under "_resolve_artifact_size()`